### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by knitHacker**
Can install it locally and import the module successfully.

### CODE REVIEW

File: tests/test_image_adding.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import pytest
```
Comment (markrawls) [inline tests/test_image_adding.py:1]: Not a change request, but +200 points to you for using `pytest` and including unit tests in v1

File: signhere/utils.py [hunk 2]
```diff
@@ -0,0 +1,277 @@
+import re
+from io import BytesIO
+from box import Box
+from fitz import Document, Rect
+
+from .exceptions import AddImageArgumentError
+
+
+DEFAULT_SCALAR = 0.75
```
Comment (markrawls) [inline signhere/utils.py:9]: Dumb question - what's this for?

**Review by markrawls — CHANGES_REQUESTED** [review]
Also I'm about 80% confident we can't (and 100% certain we shouldn't) legally host the Century Expanded Bold font file here. That and I don't think the general public using this library would be needing that. Overall though, coming along nicely!!

File: signhere/utils.py (hunk 2 above)
Comment (cviveiro) [reply signhere/utils.py:9]: i found that this was the scaling needed in the x and y direction to get things placed correctly. this is scaling from the (x, y) coords we get from pdf gen to the (x, y) coords we need to use to place the picture in the right place on the pdf

File: README.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+```bash
+pip install --upgrade sign-here
+```
+
+## Usage
+
+```python
+import signhere.utils
+signhere.utils.
```
[the remark this answers is not in the document]
Comment (knitHacker) [reply README.md:27]: haha sorry i didn't actually know how to use it so i stopped in the middle of the example.

**Comment by knitHacker** [comment]
Thanks! Will remove the font and make suggested code changes. So far I hadn't really touched the code other than update the import in tests.

File: tests/test_image_adding.py (hunk 1 above)
Comment (knitHacker) [reply tests/test_image_adding.py:1]: hahaha don't give me credit for it. it was with the initial code in doc-gen

File: signhere/utils.py [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+
+
+def add_images_to_pdf(
+    initial_pdf,
+    metadata,
+    img_loader,
+    dynamic_text,
+    only_matches=None,
+    page_numbers=None,
+    loader_args=None,
```
[the remark this answers is not in the document]
Comment (knitHacker) [reply signhere/utils.py]: do you want type annotations on all function arguments or just the public ones?

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (knitHacker) [reply README.md:27]: just to clarify you want to ditch the README.md and substitute something generated by sphinx? so I would check in the generate documentation?

**Comment by knitHacker** [comment]
@markrawls would appreciate another look through. I cleaned up the code / added in sphinx type docstrings and added in sphinx docs. Still have the readme because not sure how / if I should use generated (and checked in) sphinx docs for the readme

File: signhere/utils.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (markrawls) [reply signhere/utils.py]: Ideally all of them, but if it becomes too cumbersome then we can ditch annotating private members.

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (markrawls) [reply README.md]: Yep! I'd say check in the source files and set up a ReadTheDocs page for it, don't commit the generated documentation.

File: docs/conf.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+# import os
+# import sys
+# sys.path.insert(0, os.path.abspath('.'))
+
+
+# -- Project information -----------------------------------------------------
+
+project = "signhere"
+copyright = "2021, Mark Rawls"
+author = "Mark Rawls"
```
Comment (markrawls) [inline docs/conf.py:22]: This should probably be your name (or the company's) instead of mine

**Review by markrawls — CHANGES_REQUESTED** [review]
Outside of that, looks good!!

File: docs/conf.py (hunk 5 above)
Comment (knitHacker) [reply docs/conf.py:22]: I can put the companies name but all the actual code was written by you and christine wrote all the actual code haha

**Comment by knitHacker** [comment]
Glad I went to add type annotations to the private functions because I realized I didn't remove the references to the IMAGE_CONSTANTS dict that I removed.

**Review by markrawls — APPROVED** [review]
This is beautiful. Great job on that! We should make a ReadTheDocs org (if that's a thing they offer) and get our docs for this hosted there, but this is more than enough to get started.

**Pull request merged**